src/api/global-api.md

```diff
@@ -193,6 +193,100 @@ const AsyncComp = defineAsyncComponent({
 
 **See also**: [Dynamic and Async components](../guide/component-dynamic-async.html)
 
+## resolveComponent
+
+:::warning
+`resolveComponent` can only be used within render functions.
+:::
+
+Allows resolving a `component` by its name, if it is available in the current application instance.
+
+Returns a `Component` or `undefined` when not found.
+
+```js
+const app = Vue.createApp({})
+app.component('MyComponent', {
+  /* ... */
+})
+```
+
+```js
+import { resolveComponent } from 'vue'
+setup() {
+  const MyComponent = resolveComponent('MyComponent')
+}
+```
+
+## resolveDirective
+
+:::warning
+`resolveDirective` can only be used within render functions.
+:::
+
+Allows resolving a `directive` by its name, if it is available in the current application instance.
+
+Returns a `Directive` or `undefined` when not found.
+
+```js
+const app = Vue.createApp({})
+app.directive('highlight', {})
+```
+
+```js
+import { resolveDirective } from 'vue'
+setup () {
+  const highlightDirective = resolveDirective('highlight')
+}
+```
+
+#### Arguments
+
+- `{ String } name`
+
+## withDirectives
+
+:::warning
+`withDirectives` can only be used within render functions.
+:::
+
+Allows applying directives to a **VNode**. Returns a VNode with the applied directives.
+
+```js
+import { withDirectives, resolveDirective } from 'vue'
+const foo = resolveDirective('foo')
+const bar = resolveDirective('bar')
+
+return withDirectives(h('div'), [
+  [foo, this.x],
+  [bar, this.y]
+])
+```
+
+#### Arguments
+
+Accepts two arguments: `vnode` and `directives`.
+
+##### vnode
```

Comment:
**nitpick**: I'm afraid it will be _too_ nested 😢  Could we follow the same format as for previous arguments?

```
- `{ VNode } vnode` - a virtual node, usually created with `h()`
- `{ Array  }directives ` - an array of directives. Each directive itself is an array, which allows for up to 4 indexes to be defined as follows
...
```

Reply:
I made a mistake with the indentation actually, thanks for pointing it out.